ktor-http/common/src/io/ktor/http/auth/HttpAuthHeader.kt

```diff
@@ -42,44 +42,123 @@ public fun parseAuthorizationHeader(headerValue: String): HttpAuthHeader? {
         return HttpAuthHeader.Parameterized(authScheme, emptyList())
     }
 
-    val token68 = matchToken68(headerValue, index)
-    if (token68 != null) {
-        return HttpAuthHeader.Single(authScheme, token68)
+    val token68EndIndex = matchToken68(headerValue, index)
+    val token68 = headerValue.substring(index until token68EndIndex).trim()
+    if (token68.isNotEmpty()) {
+        if (token68EndIndex == headerValue.length) {
+            return HttpAuthHeader.Single(authScheme, token68)
+        }
     }
 
-    val parameters = matchParameters(headerValue, tokenStartIndex)
-    return HttpAuthHeader.Parameterized(authScheme, parameters)
+    val parameters = mutableMapOf<String, String>()
+    val endIndex = matchParameters(headerValue, index, parameters)
+    return if (endIndex == -1) HttpAuthHeader.Parameterized(authScheme, parameters) else
+        throw ParseException("Function parseAuthorizationHeader can parse only one header")
+}
+
+/**
+ * Parses an authorization header [headerValue] into a list of [HttpAuthHeader].
+ * @return a list of [HttpAuthHeader]
+ * @throws [ParseException] on invalid header
+ */
+public fun parseAuthorizationHeaders(headerValue: String): List<HttpAuthHeader> {
+    var index = 0
+    val headers = mutableListOf<HttpAuthHeader>()
+    while (index != -1) {
+        index = parseAuthorizationHeader(headerValue, index, headers)
+    }
+    return headers
 }
 
-private fun matchParameters(headerValue: String, startIndex: Int): Map<String, String> {
-    val result = mutableMapOf<String, String>()
+private fun parseAuthorizationHeader(
```

Comment:
Can we return an iterator here?

Reply:
The logic of parsing here previously has been done with string indexing, so i just followed it

Reply:
agree, let's keep index here